examples/mass_example/main.cpp

```diff
@@ -0,0 +1,50 @@
+#include <chrono>
+#include <iostream>
+#include <system_error>
+
+#include <tfc/ipc.hpp>
+#include <tfc/ipc/details/dbus_client_iface_mock.hpp>
+#include <tfc/progbase.hpp>
+
+#include <mp-units/systems/si/si.h>
+#include <boost/asio.hpp>
+
+namespace asio = boost::asio;
+
+int main(int argc, char** argv) {
+  tfc::base::init(argc, argv);
+
+  auto ctx{ asio::io_context() };
+  tfc::ipc_ruler::ipc_manager_client_mock ipc_client_{ ctx };
+
+  tfc::ipc::signal<tfc::ipc::details::type_mass, tfc::ipc_ruler::ipc_manager_client_mock&> weight_signal{
+    ctx,
+    ipc_client_,
+    "weight",
+  };
+
+  tfc::ipc::slot<tfc::ipc::details::type_mass, tfc::ipc_ruler::ipc_manager_client_mock&> weight_slot{
+    ctx, ipc_client_, "weight", [](const tfc::ipc::details::mass_t&) {}
+  };
+
+  ipc_client_.connect(ipc_client_.slots_[0].name, ipc_client_.signals_[0].name, [](std::error_code const&) {});
+
+  weight_signal.send(100 * mp_units::si::gram);
+
+  ctx.run_for(std::chrono::milliseconds(10));
```

Comment:
looks like ping pong test, It would be awesome to convert this to a ipc test using parameterized test.

something like:
```cpp
template <typename type_decl>
struct data_t {
  using type_description = type_decl;
  type_decl::value_t value{};
};
"ping pong"_test = [](auto& data) {
  using type_description = std::remove_ref_t<decltype(data)>::type_description;
  // make signal and slot connect and send and receive and expect upon its value
} | std::tuple{ data_t<tfc::ipc::details::type_mass>{.value = 100 * mp_units::si::gram }  }; // add all types here with its according value
```

Reply:
I added this test but that doesn't seem to test the failing piece of code, still nice to have.